exporter/kafkaexporter/pdata_marshaler.go

```diff
@@ -83,16 +86,25 @@ type pdataTracesMarshaler struct {
 }
 
 func (p pdataTracesMarshaler) Marshal(td pdata.Traces, topic string) ([]*sarama.ProducerMessage, error) {
-	bts, err := p.marshaler.MarshalTraces(td)
-	if err != nil {
-		return nil, err
-	}
-	return []*sarama.ProducerMessage{
-		{
+	var messages []*sarama.ProducerMessage
+	var errs error
+	for _, batch := range batchpersignal.SplitTraces(td) {
+		bts, err := p.marshaler.MarshalTraces(td)
+		if err != nil {
+			// continue to process batches that can be serialized
+			errs = multierr.Append(errs, err)
+			continue
+		}
+
+		// every batch should have at least one span
+		traceID := batch.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(0).TraceID().Bytes()
+		messages = append(messages, &sarama.ProducerMessage{
 			Topic: topic,
+			Key:   sarama.ByteEncoder(traceID[:]),
```

Comment:
It would be awesome if this could work similar to this https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/groupbyattrsprocessor

Reply:
Not sure I follow 100% what you are after here.

The aim of this PR is to bring it inline with the other marshaller, then follow up with a PR that supports configuration.